### PR TITLE
get decision date based on decision issues

### PR DESCRIPTION
**Pull request by jiahuang**
Connects #8031

Tasks don't have decision date, so find it based off of the decision issue. Decision issue dates are saved as UTC and we convert to local time in the browser, which throws off the date. Modified the date format to be able to get a date from UTC time.

### CODE REVIEW

File: app/controllers/decision_reviews_controller.rb [hunk 1]
```diff
@@ -66,7 +66,7 @@ def business_line
   def decision_date
     return unless task.instance_of? DecisionReviewTask
 
-    Date.parse(allowed_params.require("decision_date")).to_datetime
+    Date.parse(allowed_params.require("decision_date")).in_time_zone.utc.to_datetime
```
Comment (shanear) [inline app/controllers/decision_reviews_controller.rb]:
```
.in_time_zone.utc
```

this seems like too many things. Shouldn't `.utc` do the same thing?

File: client/app/nonComp/components/Disposition.jsx [hunk 2]
```diff
@@ -138,6 +138,12 @@ class NonCompDispositions extends React.PureComponent {
 
     let completeDiv = null;
 
+    let decisionDate = this.state.decisionDate;
+
+    if (appeal.decisionIssues.length > 0) {
+      decisionDate = formatDateStrUtc(appeal.decisionIssues[0].promulgationDate);
```
Comment (shanear) [inline client/app/nonComp/components/Disposition.jsx]: might want to check with sally on whether we are using `promulgation_date` for non-comp issues

File: client/app/util/DateUtil.js [hunk 3]
```diff
@@ -18,16 +18,18 @@ export const formatDate = function(dateString) {
...
+  const day = StringUtil.leftPad(date.getDate(), 2, '0');
+  const year = date.getFullYear();
 
   return `${month}/${day}/${year}`;
 };
 
 // Date format YYYY-MM-DD
-export const formatDateStr = (dateString, dateFormat = 'YYYY-MM-DD', expectedFormat = dateFormatString) => {
+export const formatDateStr = (dateString, dateFormat = 'YYYY-MM-DD',
+  expectedFormat = dateFormatString, utc = false) => {
```
Comment (pkarman) [inline client/app/util/DateUtil.js]: it's not clear to me what the `utc` param means here. Does it mean that `dateString` is in the UTC timezone? Or that it should be converted to the UTC timezone?

File: client/app/util/DateUtil.js [hunk 4]
```diff
@@ -43,7 +45,17 @@ export const formatDateStr = (dateString, dateFormat = 'YYYY-MM-DD', expectedFor
     }
   }
 
-  return moment(dateString, dateStringFormat).format(expectedFormat);
+  let date = moment(dateString, dateStringFormat);
+
+  if (utc) {
+    date = date.utc();
```
Comment (pkarman) [inline client/app/util/DateUtil.js]: this converts the timezone? does it also shift the time?

File: spec/feature/non_comp/dispositions_spec.rb [hunk 5]
```diff
@@ -3,6 +3,7 @@
 feature "NonComp Dispositions Task Page" do
   before do
     FeatureToggle.enable!(:decision_reviews)
+    Timecop.freeze(Time.zone.today)
```
Comment (pkarman) [inline spec/feature/non_comp/dispositions_spec.rb]: the browser time + zone will not reflect this - does that matter?

File: spec/feature/non_comp/dispositions_spec.rb [hunk 6]
```diff
@@ -86,13 +87,13 @@ def find_disabled_disposition(num, disposition, description = nil)
...
     scenario "saves decision issues" do
       visit dispositions_url
-
       expect(page).to have_button("Complete", disabled: true)
 
       # set description & disposition for each request issue
       fill_in_disposition(0, "Granted")
       fill_in_disposition(1, "Granted", "test description")
       fill_in_disposition(2, "Denied", "denied")
+      fill_in "decision-date", with: "01/01/2019"
```
Comment (pkarman) [inline spec/feature/non_comp/dispositions_spec.rb]: this date is unknown to me - is it just arbitrary?

File: client/app/util/DateUtil.js (hunk 3 above)
Comment (jiahuang) [reply client/app/util/DateUtil.js]: It means that the datestring being passed in doesn't have a timezone attached to it, so moment will attempt to convert to the local timezone unless utc conversion is specified.

File: client/app/util/DateUtil.js (hunk 4 above)
Comment (jiahuang) [reply client/app/util/DateUtil.js]: Correct, the time is shifted to be the same time but in UTC.

File: client/app/util/DateUtil.js (hunk 3 above)
Comment (pkarman) [reply client/app/util/DateUtil.js]: ah ok. Maybe `forceUTC` would be clearer?

**Pull request merged**